diff -up lz4-0.7.0/setup.py.orig lz4-0.7.0/setup.py
--- lz4-0.7.0/setup.py.orig	2015-06-27 18:13:28.886225670 +0100
+++ lz4-0.7.0/setup.py	2015-06-27 22:05:45.273399572 +0100
@@ -17,17 +17,12 @@ setup(
     package_dir={'': 'src'},
     ext_modules=[
         Extension('lz4', [
-            'src/lz4.c',
-            'src/lz4hc.c',
             'src/python-lz4.c'
         ], extra_compile_args=[
             "-std=c99",
-            "-O3",
-            "-Wall",
-            "-W",
-            "-Wundef",
-            "-DLZ4_VERSION=\"r119\"",
-        ])
+        ], libraries=[
+            "lz4",
+        ],)
     ],
     setup_requires=["nose>=1.0"],
     test_suite = "nose.collector",
